iphone/Maps/Classes/MapsAppDelegate.mm

```diff
@@ -556,6 +554,9 @@ - (void)applicationWillResignActive:(UIApplication *)application
 {
   [self.mapViewController.appWallAd close];
   [RouteState save];
+  
+  [self.locationManager onBackground];
+  [self.mapViewController onEnterBackground];
```

Comment:
@igrechuhin Читаем внимательно документацию:

In your app delegate’s applicationDidEnterBackground: <https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIApplicationDelegate_Protocol/index.html#//apple_ref/occ/intfm/UIApplicationDelegate/applicationDidEnterBackground:> method, your app may want to delete some of its OpenGL ES objects to make memory and resources available to the foreground app. Call the glFinish function to ensure that the resources are removed immediately.

After your app exits its applicationDidEnterBackground: <https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIApplicationDelegate_Protocol/index.html#//apple_ref/occ/intfm/UIApplicationDelegate/applicationDidEnterBackground:> method, it must not make any new OpenGL ES calls. If it makes an OpenGL ES call, it is terminated by iOS.


Удаление ресурсов из didEnterBackground через вызовы OpenGL методов - это нормальная документированная практика. К слову, именно так наш код сейчас и делает - вызывает всякие openGL методы прямо в didEnterBackground.

Нельзя вызывать OpenGL код ПОСЛЕ того, как вернулось управление системе из didEnterBackground.


> On Mar 21, 2016, at 13:21, Илья Гречухин <notifications@github.com> wrote:
> 
> In iphone/Maps/Classes/MapsAppDelegate.mm <https://github.com/mapsme/omim/pull/2365#discussion_r56801829>:
> 
> > @@ -556,6 +554,9 @@ - (void)applicationWillResignActive:(UIApplication *)application
> >  {
> >    [self.mapViewController.appWallAd close];
> >    [RouteState save];
> > +  
> > +  [self.locationManager onBackground];
> > +  [self.mapViewController onEnterBackground];
> applicationDidEnterBackground - уже поздно
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2365/files/4c1ad60019dfa841acbca6cd814eb2661ea98b14#r56801829>